docs/guide/media.md

```diff
@@ -0,0 +1,167 @@
+
+```javascript
+// modules/article/index.js
+module.exports = {
+  // ...
+  fields: {
+    add: {
+      fileUpload: {
+        label: 'File upload',
+        type: 'attachment',
+        fileGroup: 'office'
+      }
+    }
+  }
+};
+```
+
+![an attachment field](/images/media-attachment.png)
+
+### Using attachment field values in templates
+
+Getting file information from an attachment field value is simpler than the other options since you have immediate access to the attachment object. If the field on a piece type is called `fileUpload`, the attachment object is `data.piece.fileUpload` in a show page template.
+
+For image attachments, there will be an object of URLs on the `_urls` property. Apostrophe generates multiple sizes of each uploaded image as discussed [regarding the image widget](/guide/core-widgets.md#image-widget). For "office" files (e.g., PDFs) there is one `_url` property.
+
+In either case, there is a helper method available in templates to retrieve the url: `apos.attachment.url()`. For "office" attachments, simply pass the attachment object into the method.
+
+```django
+{# modules/article-page/views/show.html #}
+{% set fileUrl = apos.attachment.url(data.piece.fileUpload) %}
+
+<a href="{{ fileUrl }}">Download</a>
+```
+
+For image attachments, doing the same thing will return the URL for the `full` image size by default. You can pass an options object as a second argument with its `size` property set to another image size to get a different URL back.
+
+```django
+{# modules/article-page/views/show.html #}
+
+{% set imgUrl = apos.attachment.url(data.piece.photoUpload, {
+  size: 'one-third'
+}) %}
+
+<img src="{{ imgUrl }}" alt="" />
+```
+
+[Responsive images](https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images) are very important for cross-device support these days. Getting each image size to populate the `srcset` attribute for an image would get repetitive very quickly even using the `apos.attachment.url()` method. For that, there is a dedicated `apos.image.srcset()` method. Pass in the attachment object as an argument and it will return a `srcset` value with all sizes included.
+
+```django
+{# modules/article-page/views/show.html #}
+{% set srcset = apos.attachment.srcset(data.piece.photoUpload) %}
+
+<img srcset="{{ srcset }}" src="{{ apos.attachment.url(data.piece.photoUpload) }}" alt="" />
+```
+
+Notice that in the examples above **the alt text is not included when using an attachment field**. Since alt text should be populated in most all instances, the other two options will usually be better for images.
+
+## The image widget option
```

Comment:
Another example of where this should be synced up with the list at the top of the page.

Reply:
Also, maybe this one should come first as it's the simplest?